apps/picsa-tools/manual-tool/src/app/pages/activity-details/activity-details.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+import { IManualActivity } from '../../data/manual-contents';
+
+@Component({
+  selector: 'picsa-manual-activity-details',
+  templateUrl: './activity-details.component.html',
+  styleUrls: ['./activity-details.component.scss'],
+})
+export class ActivityDetailsComponent implements OnInit {
+  activity: IManualActivity;
+
+  constructor(private route: ActivatedRoute) {}
+
+  ngOnInit() {
+    this.activity = history.state.activity;
```

Comment:
nit(non-blocking)
Passing the activity using history state will mean the page will only work when navigated to via link, but it might also be nice to work directly (e.g. someone shares a url directly to the activity page)
A better way to implement might be to use the route `paramMap` to identify the activity id, and retrieve from the list of all activities

Reply:
Yes I get your point I hadn't thought of that use case